File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        book = {}
        for i, v in enumerate(nums):
            diff = target - v
            if(diff in book):
                return [book[diff], i]
            else:
                book[v] = i

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int r = 0;
        int l = 0;
        int sum = 0;
        int min = nums.length;
        
        while(r < nums.length){
            sum += nums[r];
            
            while(sum >= target){
                min = Math.min(min, r-l+1);
                sum -= nums[l];
                l++;
            }

            r++;
        }

        if(r - l == nums.length && sum < target) return 0;
        else return min;
    }
}